x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -1162,22 +1416,64 @@ public void testPushDownLimit_PastEvalAndMvExpand() {
         assertThat(orderNames(topN), contains("salary"));
         var eval = as(topN.child(), Eval.class);
         var mvExp = as(eval.child(), MvExpand.class);
-        topN = as(mvExp.child(), TopN.class);
-        assertThat(topN.limit().fold(), equalTo(10000));
+        as(mvExp.child(), EsRelation.class);
+    }
+
+    /**
+     * Expected
+     * TopN[[Order[salary{f}#11,ASC,LAST]],5[INTEGER]]
+     * \_MvExpand[first_name{f}#7,first_name{r}#16]
+     *   \_EsRelation[test][_meta_field{f}#12, emp_no{f}#6, first_name{f}#7, ge..]
+     */
+    public void testPushDownMvExpand_PastSameFieldSorting2() {
+        LogicalPlan plan = optimizedPlan("""
+            from test
+            | sort first_name
+            | mv_expand first_name
```

Comment:
I think here we could duplicate the topn and put it before the MV expand.

Reply:
If you refer to topn(`salary`) yes, probably. This was my first attempt at mv_expand planning (an older, different PR). But `sort first_name` needs to go away as well because it has no effect on the results, nor on what happens after it.

Reply:
Sorry, should have marked my comment more clearly: this is just a thought for future optimizations that I wanted to leave. (Out of scope of this PR)